spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/CometColumnReader.java

```diff
@@ -0,0 +1,164 @@
+
+import java.io.IOException;
+import java.util.Map;
+import org.apache.comet.parquet.AbstractColumnReader;
+import org.apache.comet.parquet.ColumnReader;
+import org.apache.comet.parquet.TypeUtil;
+import org.apache.comet.parquet.Utils;
+import org.apache.iceberg.parquet.VectorizedReader;
+import org.apache.iceberg.spark.SparkSchemaUtil;
+import org.apache.iceberg.types.Types;
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.parquet.column.page.PageReadStore;
+import org.apache.parquet.column.page.PageReader;
+import org.apache.parquet.hadoop.metadata.ColumnChunkMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnPath;
+import org.apache.spark.sql.internal.SQLConf;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+
+/**
+ * A Iceberg Parquet column reader backed by a Comet {@link ColumnReader}. This class should be used
+ * together with {@link CometVector}.
+ *
+ * <p>Example:
+ *
+ * <pre>
+ *   CometColumnReader reader = ...
+ *   reader.setBatchSize(batchSize);
+ *
+ *   while (hasMoreRowsToRead) {
+ *     if (endOfRowGroup) {
+ *       reader.reset();
+ *       PageReader pageReader = ...
+ *       reader.setPageReader(pageReader);
+ *     }
+ *
+ *     int numRows = ...
+ *     CometVector vector = reader.read(null, numRows);
+ *
+ *     // consume the vector
+ *   }
+ *
+ *   reader.close();
+ * </pre>
+ */
+@SuppressWarnings({"checkstyle:VisibilityModifier", "ParameterAssignment"})
+class CometColumnReader implements VectorizedReader<CometVector> {
+  public static final int DEFAULT_BATCH_SIZE = 5000;
+
+  private final DataType sparkType;
+  protected AbstractColumnReader delegate;
+  private final CometVector vector;
+  private final ColumnDescriptor descriptor;
+  protected boolean initialized = false;
+  protected int batchSize = DEFAULT_BATCH_SIZE;
+
+  CometColumnReader(DataType sparkType, ColumnDescriptor descriptor) {
+    this.sparkType = sparkType;
+    this.descriptor = descriptor;
+    this.vector = new CometVector(sparkType, false);
+  }
+
+  CometColumnReader(Types.NestedField field) {
+    DataType dataType = SparkSchemaUtil.convert(field.type());
+    StructField structField = new StructField(field.name(), dataType, false, Metadata.empty());
+    this.sparkType = dataType;
+    this.descriptor = TypeUtil.convertToParquet(structField);
+    this.vector = new CometVector(sparkType, false);
+  }
+
+  public AbstractColumnReader getDelegate() {
+    return delegate;
+  }
+
+  /**
+   * This method is to initialized/reset the ColumnReader. This needs to be called for each row
+   * group after readNextRowGroup, so a new dictionary encoding can be set for each of the new row
+   * groups.
+   */
+  public void reset() {
+    if (delegate != null) {
+      delegate.close();
+    }
+
+    delegate = Utils.getColumnReader(sparkType, descriptor, batchSize, SQLConf.get());
```

Comment:
Just out of my curiosity, why not call `SQLConf.get()` inside the method?

Reply:
Right, this should be called inside the method